src/hypofuzz/hy.py

```diff
@@ -330,22 +337,31 @@ def _run_test_on(
             self.since_new_cov = 0
         else:
             self.since_new_cov += 1
-        if 0 in (self.since_new_cov, self.ninputs % 100):
-            UNDELIVERED_REPORTS.append(self._json_description)
+        if 0 in (self.since_new_cov, self.ninputs % 100) and not self._replay_buffer:
+            self._reports.append(self._json_description)
 
         self.elapsed_time += time.perf_counter() - start
-        if UNDELIVERED_REPORTS and (self._last_post_time + 10 < self.elapsed_time):
-            self._report_change(UNDELIVERED_REPORTS)
-            del UNDELIVERED_REPORTS[:]
+        if self._last_post_time + 10 < self.elapsed_time:
+            self._report_changes()
```

Comment:
Given the changes to report delivery, in particular that we can naturally batch http requests but can't batch database writes, I think we should make a couple of changes here.

- report changes immediately, rather than waiting 10s
- move the actual database logic to a background thread, so that the fuzzing worker just drops the message in a queue and moves on.  
  - Plausibly this should take the form of an `InBackgroundThreadDatabase(db)` wrapper which can live in Hypothesis itself?  Fuzzing can't be the only usecase, and doing some smart stuff like issuing writes before deletes in case of shutdown would be pretty easy to handle in one place.  ("database is only ever best-effort" is such a nice invariant)
- for the `self.ninputs % 100 == 0` case, we should track the previous report and if nothing has changed, drop it from the database.  This is mostly a win for the plotting code, which gets unhappy with very large numbers of points, but also analysis in general.

unrelatedly it'd be nice to have a stats-as-json endpoint that users can hit to get all the raw data we're displaying, in case they want to slice and maybe display it for themselves 🤔